test/test_type_inference.py

```diff
@@ -0,0 +1,43 @@
+import pytest
+
+from acclimatise.cli_types import (
+    CliBoolean,
+    CliDir,
+    CliFile,
+    CliFloat,
+    CliInteger,
+    CliString,
+)
+from acclimatise.model import infer_type
+
+
+@pytest.mark.parametrize(
+    "string,typ,isoutput",
+    [
+        ("", CliString, False),
+        ("int", CliInteger, False),
+        ("size", CliInteger, False),
+        ("length", CliInteger, False),
+        ("max", CliInteger, False),
+        ("min", CliInteger, False),
+        ("str", CliString, False),
+        ("float", CliFloat, False),
+        ("decimal", CliFloat, False),
+        ("bool", CliBoolean, False),
+        ("file", CliFile, None),
+        ("path", CliFile, None),
+        ("input file", CliFile, False),
+        ("output file", CliFile, True),
+        ("folder", CliDir, None),
+        ("directory", CliDir, None),
+        ("output directory", CliDir, True),
+        ("blah 23 blub", CliInteger, False),
+        ("nonsense 23.42", CliFloat, True),
+        (".42 gibberish", CliFloat, True),
+        ("1E-5", CliFloat, True),
+    ],
+)
+def test_type_inference(string, typ, isoutput):
```

Comment:
Actually I wonder if the tests could be simpler if you just had the input string, and then the expected output, ie:
```
  "string,expected",
    [
        ("", CliString()),
        ("int", CliInteger()),
        ("size", CliInteger()),
        ("output file", CliFile(output=True)),
        ("folder", CliDir(output=False)),
    ]
```
And then you just `assert infer_type(string) == expected`

Reply:
Indeed. Consider it done.